deploy-to-aks/action.yml

```diff
@@ -0,0 +1,147 @@
+name: Deploy to AKS environment
+description: Deploy a docker image to an AKS environment
+
+inputs:
+  azure-credentials:
+    required: true
+  environment:
+    description: Environment to deploy to
+    required: true
+  github-token:
+    description: Default Github token retrieved via secrets.GITHUB_TOKEN or PAT with permission to the repository (Required)
+    required: true
+  pr-number:
+    description: PR number for the review app
+    required: false
+  sha:
+    description: Commit sha to be deployed
+    required: true
+  slack-webhook:
+    required: false
+  db-seed:
+    description: run seed command after a review deployment
+    type: boolean
+    required: false
+    default: false
+  smoke-test:
+    description: Enable smoke test after deployment
+    type: boolean
+    required: false
+    default: false
+  healthcheck:
+    description: Health check path
+    required: false
+  terraform-url-output:
+    description: terraform output that contains the app urls
+    required: false
+    default: 'external_urls'
+  terraform-base:
+    description: path to the terraform files
+    required: false
+    default: 'terraform/application'
+  terraform-provider:
+    description: file containing terraform provider
+    required: false
+    default: 'terraform.tf'
+
+runs:
+  using: composite
+  steps:
+    - name: Set Environment variables
+      id: set_env_var
+      shell: bash
+      run: |
+        terraform_version=$(awk '/{/{f=/^terraform/;next}f' ${{ inputs.terraform-base }}/${{ inputs.terraform-provider }} | grep -o [0-9\.]*)
+        echo "TERRAFORM_VERSION=$terraform_version" >> $GITHUB_ENV
+        echo "GCP_PROJECT_ID=$(make -s printvar-GCP_PROJECT_ID || true)" >> $GITHUB_ENV
+        echo "GCP_WIP=$(make -s printvar-GCP_WIP || true)" >> $GITHUB_ENV
```

Comment:
This long string is only used for workflows. To simplify, could we try a reposiroty variable instead? https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#creating-configuration-variables-for-a-repository

Reply:
Or pass it as input as discussed before